nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java

```diff
@@ -809,6 +809,12 @@ public void schedule(Work work, Scheduling scheduling, boolean afterCommit) {
             break;
 
         }
+        if (work.isGroupJoin()) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Submit Work: %s to GroupJoin: %s", work.getId(), work.getPartitionKey()));
```

Comment:
You could use log4j2 here:
```suggestion
                log.debug("Submit Work: {} to GroupJoin: {}", work::getId, work::getPartitionKey);
```